its/src/org/sonarlint/eclipse/its/QuickFixesTest.java

```diff
@@ -0,0 +1,186 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.eclipse.its;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.eclipse.reddeer.common.wait.WaitUntil;
+import org.eclipse.reddeer.core.matcher.WithLabelMatcher;
+import org.eclipse.reddeer.eclipse.core.resources.Project;
+import org.eclipse.reddeer.eclipse.core.resources.Resource;
+import org.eclipse.reddeer.eclipse.ui.dialogs.PropertyDialog;
+import org.eclipse.reddeer.eclipse.ui.perspectives.JavaPerspective;
+import org.eclipse.reddeer.eclipse.ui.views.markers.ProblemsView;
+import org.eclipse.reddeer.eclipse.ui.views.markers.ProblemsView.ProblemType;
+import org.eclipse.reddeer.eclipse.ui.views.markers.QuickFixWizard;
+import org.eclipse.reddeer.jface.text.contentassist.ContentAssistant;
+import org.eclipse.reddeer.swt.impl.button.PushButton;
+import org.eclipse.reddeer.swt.impl.menu.ContextMenuItem;
+import org.eclipse.reddeer.swt.impl.table.DefaultTable;
+import org.eclipse.reddeer.swt.impl.text.DefaultText;
+import org.eclipse.reddeer.workbench.condition.TextEditorContainsText;
+import org.eclipse.reddeer.workbench.impl.editor.TextEditor;
+import org.eclipse.reddeer.workbench.ui.dialogs.WorkbenchPreferenceDialog;
+import org.junit.After;
+import org.junit.Test;
+import org.sonarlint.eclipse.its.reddeer.preferences.SonarLintPreferences;
+import org.sonarlint.eclipse.its.reddeer.preferences.SonarLintPreferences.MarkerSeverity;
+import org.sonarlint.eclipse.its.reddeer.views.OnTheFlyView;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class QuickFixesTest extends AbstractSonarLintTest {
+  private static final String QUICK_FIX_MESSAGE = "Remove the cast to \"int\"";
+  private static final String EXPECTED_TEXT = "return hashCode();";
+  private static final String FILE_NAME = "Hello.java";
+  private static final String PLACEHOLDER_LICENSE_HEADER = "/**\nPlaceholder\nlicense\nheader\n*/";
+
+  @After
+  public void tearDown() {
+    // default value
+    setMarkerSeverityToWarning(MarkerSeverity.INFO);
+  }
+
+  @Test
+  public void shouldApplyQuickFixThroughContentAssist() {
+    importAndOpenQuickFixableFile();
+
+    TextEditor editor = new TextEditor(FILE_NAME);
+    editor.setCursorPosition(8, 14);
+    // uses the keyboard and may not work on non-US layouts
+    ContentAssistant assistant = editor.openQuickFixContentAssistant();
+    assertThat(assistant.getProposals()).element(0).isEqualTo(QUICK_FIX_MESSAGE);
+    assistant.chooseProposal(QUICK_FIX_MESSAGE);
+
+    new WaitUntil(new TextEditorContainsText(editor, EXPECTED_TEXT));
+  }
+
+  @Test
+  public void shouldApplyQuickFixThroughMarkerContextMenu() {
+    // default issue severity is INFO and reddeer supports only ERROR and WARNING
```

Comment:
Can't we use `ProblemType.ALL`?

Reply:
It wouldn't work for INFO : https://github.com/eclipse/reddeer/blob/master/plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/ui/views/markers/ProblemsView.java#L46